pkg/resourcemanager/azuresql/azuresqluser/azuresqluser_reconcile.go

```diff
@@ -234,19 +240,24 @@ func (s *AzureSqlUserManager) Ensure(ctx context.Context, obj runtime.Object, op
 					formattedSecrets["username"] = []byte(user)
 
 				case "password":
-					formattedSecrets["password"] = DBSecret["password"]
+					formattedSecrets["password"] = userSecret["password"]
 				}
 			} else {
-				err = sqlUserSecretClient.Delete(
```

Comment:
Same comment here -- it looks as if this error was previously just ignored, which may be important if customers have blocked deletes?